From 69f8dd358cb78c7a10d2b2b5b75879d2d3b78c68 Mon Sep 17 00:00:00 2001
From: jianzhou <jianzhou@codeaurora.org>
Date: Tue, 4 Mar 2014 21:45:52 -0500
Subject: [PATCH 1/4] Settings: Add charge mode for usb settings

Add charge mode for usb settings, once select "Charge mode", property
"persist.sys.usb.config" will be set to "charging", device only can be
charged.

Conflicts:
res/values/strings.xml
Change-Id: I6d88ad8201f2f52be948e07fd1e9de9b4caf808d

Conflicts:
	res/values/strings.xml
---
 res/values/strings.xml                               |  2 ++
 res/xml/usb_settings.xml                             |  6 ++++++
 src/com/android/settings/deviceinfo/UsbSettings.java | 15 ++++++++++++++-
 3 files changed, 22 insertions(+), 1 deletion(-)

diff --git a/res/values/strings.xml b/res/values/strings.xml
index dac2862..27bb819 100644
--- a/res/values/strings.xml
+++ b/res/values/strings.xml
@@ -6068,4 +6068,6 @@
    <string name="imei_information_summary">IMEI relative information</string>
    <string name="slot_number">(Slot<xliff:g id="slot_num">%1$d</xliff:g>)</string>
 
+   <string name="usb_charging_title">Charging</string>
+   <string name="usb_charging_summary">Lets you disable all data and file transfer connection with computer and put device in charging only mode.</string>
 </resources>
diff --git a/res/xml/usb_settings.xml b/res/xml/usb_settings.xml
index 4b0268b..b78ab42 100644
--- a/res/xml/usb_settings.xml
+++ b/res/xml/usb_settings.xml
@@ -32,4 +32,10 @@
         android:summary="@string/usb_ptp_summary"
         />
 
+    <CheckBoxPreference
+        android:key="usb_charging"
+        android:title="@string/usb_charging_title"
+        android:summary="@string/usb_charging_summary"
+        />
+
 </PreferenceScreen>
diff --git a/src/com/android/settings/deviceinfo/UsbSettings.java b/src/com/android/settings/deviceinfo/UsbSettings.java
index 42de2fd..f2623c5 100644
--- a/src/com/android/settings/deviceinfo/UsbSettings.java
+++ b/src/com/android/settings/deviceinfo/UsbSettings.java
@@ -41,10 +41,12 @@ public class UsbSettings extends SettingsPreferenceFragment {
 
     private static final String KEY_MTP = "usb_mtp";
     private static final String KEY_PTP = "usb_ptp";
+    private static final String KEY_CHARGING = "usb_charging";
 
     private UsbManager mUsbManager;
     private CheckBoxPreference mMtp;
     private CheckBoxPreference mPtp;
+    private CheckBoxPreference mCharging;
     private boolean mUsbAccessoryMode;
 
     private final BroadcastReceiver mStateReceiver = new BroadcastReceiver() {
@@ -68,6 +70,7 @@ public class UsbSettings extends SettingsPreferenceFragment {
 
         mMtp = (CheckBoxPreference)root.findPreference(KEY_MTP);
         mPtp = (CheckBoxPreference)root.findPreference(KEY_PTP);
+        mCharging = (CheckBoxPreference)root.findPreference(KEY_CHARGING);
 
         UserManager um = (UserManager) getActivity().getSystemService(Context.USER_SERVICE);
         if (um.hasUserRestriction(UserManager.DISALLOW_USB_FILE_TRANSFER)) {
@@ -107,13 +110,21 @@ public class UsbSettings extends SettingsPreferenceFragment {
         if (UsbManager.USB_FUNCTION_MTP.equals(function)) {
             mMtp.setChecked(true);
             mPtp.setChecked(false);
+            mCharging.setChecked(false);
         } else if (UsbManager.USB_FUNCTION_PTP.equals(function)) {
             mMtp.setChecked(false);
             mPtp.setChecked(true);
-        } else  {
+            mCharging.setChecked(false);
+        }  else if (UsbManager.USB_FUNCTION_CHARGING.equals(function)) {
             mMtp.setChecked(false);
             mPtp.setChecked(false);
+            mCharging.setChecked(true);
+        }  else  {
+            mMtp.setChecked(false);
+            mPtp.setChecked(false);
+            mCharging.setChecked(false);
         }
+
         UserManager um = (UserManager) getActivity().getSystemService(Context.USER_SERVICE);
         if (um.hasUserRestriction(UserManager.DISALLOW_USB_FILE_TRANSFER)) {
             Log.e(TAG, "USB is locked down");
@@ -151,6 +162,8 @@ public class UsbSettings extends SettingsPreferenceFragment {
             function = UsbManager.USB_FUNCTION_MTP;
         } else if (preference == mPtp && mPtp.isChecked()) {
             function = UsbManager.USB_FUNCTION_PTP;
+        } else if (preference == mCharging && mCharging.isChecked()) {
+            function = UsbManager.USB_FUNCTION_CHARGING;
         }
 
         mUsbManager.setCurrentFunction(function, true);
-- 
1.9.1

